### PR TITLE
Multi data set fitting in indirect data analysis GUI

### LINKED ISSUE #22055: Indirect - Fitting Interfaces - No existing option to specify whether to perform a sequential or simultaneous fit
The Indirect fitting interfaces (ConvFit, IqtFit, MSDFit, JumpFit), do not currently provide an option to perform either a sequential or a simultaneous fit. In ConvFit, MSDFit and JumpFit, a sequential fit is the only possible, default option. In IqtFit a ```Make Beta Global``` option is available -- which an option between a sequential and simultaneous fit would remove.

### PR DESCRIPTION
**Pull request by mantid-roman**
**Description of work.**

Replaced the fit property browser in the indirect data analysis interface with the refactored FunctionBrowser supporting both single and multi data set fitting.

**To test:**

Fixes #22055

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by alicerussell1 — CHANGES_REQUESTED** [review]
I've started going through some of the fitting files (mainly just looked at one tab so I could follow what was going on). They look good and what you've done makes sense. The one thing I did notice was large parts of code are commented out instead of removed, I assume this is because you were referring to it when doing other parts but now it probably needs to go as it will be confusing in future.

I've gone though: IndirectFitData.cpp , ConvFit.cpp, ConvFit.h, ConvFitDataPresenter.cpp, ConvFitDataPresenter.h, ConvFunctionModel.cpp, ConvFunctionModel.h, ConvFitDataTablePresenter.cpp, ConvFitDataTablePresenter.h, ConvFitModel.cpp, ConvFitModel.h

File: qt/scientific_interfaces/Indirect/IqtFit.cpp [hunk 1]
```diff
@@ -44,64 +46,39 @@ IqtFit::IqtFit(QWidget *parent)
...
   setOutputOptionsView(m_uiForm->ovOutputOptionsView);
+  auto templateBrowser = new IqtTemplateBrowser;
+  m_uiForm->fitPropertyBrowser->setFunctionTemplateBrowser(templateBrowser);
   setFitPropertyBrowser(m_uiForm->fitPropertyBrowser);
 
   setEditResultVisible(true);
 }
 
 void IqtFit::setupFitTab() {
+  // setSampleWSSuffices({"_iqt"});
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/IqtFit.cpp]: I believe these can be removed completely

File: qt/scientific_interfaces/Indirect/JumpFitModel.cpp [hunk 2]
```diff
@@ -31,32 +32,33 @@ struct ContainsOneOrMore {
 };
 
 template <typename Predicate>
-std::pair<std::vector<std::string>, std::vector<std::size_t>>
+std::pair<std::vector<std::string>, std::vector<IDAWorkspaceIndex>>
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/JumpFitModel.cpp]: what is the difference between an `IDAWorkspaceIndex` and a `WorkspaceIndex`?

**Review by robertapplin — CHANGES_REQUESTED** [review]
I have looked through most of the files, and the changes look very good. Same as Alice, I noticed that large parts of the code are commented out. Are these to be removed, or are they yet to be re-implemented?

I had a few other minor comments below, however the changes made look very good.

**Comment by AnthonyLim23** [comment]
When @robertapplin and @alicerussell1 are happy with the code I will test the non-indirect fitting

**Comment by AnthonyLim23** [comment]
@robertapplin could we have a chat about this PR later today?

**Comment by AnthonyLim23** [comment]
file conflicts

**Comment by AnthonyLim23** [comment]
@Mantid-Matthew can we close this PR as we are splitting it up?

**Comment by NickDraper** [comment]
Removing the milestone as this is not ready, put it back if this is urgent for R4.2

**Comment by AnthonyLim23** [comment]
this has been split up by @Mantid-Matthew into smaller PR's so I am closing this

**Pull request closed without merging**